src/components/DataTable.vue

```diff
@@ -15,13 +15,21 @@
           <i
             class="fa-solid fa-pencil icon"
             @click="
-              $emit('edit', table.row);
-              openAppModal();
+              (event) => {
```

Comment:
You can use one function here and pass it `(table.row, $event)`, so the template is cleaner